cmd/tools/gendynamicconfig/main.go

```diff
@@ -168,7 +168,17 @@ func (s {{.P.Name}}TypedSetting[T]) WithDefault(v T) {{.P.Name}}TypedSetting[T]
 	return newS
 }
 
-func (s {{.P.Name}}TypedSetting[T]) Get(c *Collection) func({{.P.GoArgs}}) T {
+{{if eq .P.Name "Global" -}}
+type TypedPropertyFn[T any] func({{.P.GoArgs}}) T
+{{- else -}}
+type TypedPropertyFnWith{{.P.Name}}Filter[T any] func({{.P.GoArgs}}) T
+{{- end}}
```

Comment:
This part is in https://github.com/temporalio/temporal/pull/5962 (which still needs a review...)